### ISSUE TITLE
False positive for union case indentation when using attributes

### ISSUE THREAD

**Issue by MangelMaxime** [fsprojects/FSharpLint#292]
### Description

False positive for "Union definition cases should be at the same indentation level."

### Repro steps

Please provide the steps required to reproduce the problem

This code generated the warning mentioned:

```fs
type Option =
    | [<CompiledName("is-normal")>] IsNormal
    | CustomClass of string
```

However, if I try to put `[<CompiledName("is-normal")>]` above the `IsNormal` case to get rid of the warning then the code is invalid.

### Expected behavior

Should not generate a warning.

### Actual behavior

Generate a warning because `IsNormal` and `CustomClass` are not aligned.

### Known workarounds

No workarounds know, the only solution is to disable the setting.
